### PR TITLE
Add mill build system

### PR DESCRIPTION
**Pull request by sequencer**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**
This PR adds mill build system.
`common.sc` is used for using rocketchip as in-project build-from-source dependency.
for example(`mychisel3`, `myhardfloat`, `myconfig` are other in-project dependencies.)
```
import $`file.rocket-chip.common`

object myrocketchip extends rocketchip.common.CommonRocketChip {
  override def millSourcePath = os.pwd / "rocket-chip"

  def chisel3Module: Option[PublishModule] = Some(mychisel3)

  def hardfloatModule: PublishModule = myhardfloat

  def configModule: PublishModule = myconfig
}
```
can create a `myrocketchip` module, while other project can directly depend on it

`build.sc` is used for running rocket-chip as top project:
Using `mill rocketchip.run` to run `RocketChipStage`
Using `mill rocketchip.test` to run in-project tests.

### CODE REVIEW

File: common.sc [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import mill.{Agg, _}
+import mill.scalalib._
+import mill.scalalib.publish._
+import coursier.maven.MavenRepository
+
+val defaultVersions = Map(
+  "chisel3" -> "3.4.0-RC3",
+  "chisel3-plugin" -> "3.4.0-RC3"
+)
```
Comment (jackkoenig) [inline common.sc:9]:
```suggestion
val defaultVersions = Map(
  "chisel3" -> "3.4.0",
  "chisel3-plugin" -> "3.4.0"
)
```

File: common.sc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  private val macroParadise = ivy"org.scalamacros:::paradise:2.1.1"
+
+  private val chisel3Plugin = getVersion("chisel3-plugin")
+
+  override def repositories = super.repositories ++ Seq(
+    MavenRepository("https://oss.sonatype.org/content/repositories/snapshots"),
+    MavenRepository("https://oss.sonatype.org/content/repositories/releases")
+  )
+
+  override def compileIvyDeps = Agg(macroParadise) ++ (if(chisel3Module.isDefined) Agg() else Agg(chisel3Plugin))
```
Comment (jackkoenig) [inline common.sc:107]: Is this needed since the plugin is declared in the `scalacPluginIvyDeps`?

**Review by jackkoenig — APPROVED** [review]
Looks good with a couple of suggestions

File: common.sc (hunk 2 above)
Comment (sequencer) [reply common.sc:107]: I just checked in:
https://github.com/lihaoyi/mill/blob/cc0d0fc2de175d295e7905f11d709ca69536e616/integration/test/resources/play-json/build.sc#L70-L75, so I think `macroParadise` should exist in both cases.
But I think `chisel3Plugin` only needed in `scalacPluginIvyDeps`.

**Pull request merged**